sim/shaman/firenova.go

```diff
@@ -1,3 +1,50 @@
 package shaman
 
-// TODO: implement fire nova
+import (
+	"time"
+
+	"github.com/wowsims/wotlk/sim/core"
+	"github.com/wowsims/wotlk/sim/core/proto"
+	"github.com/wowsims/wotlk/sim/core/stats"
+)
+
+func (shaman *Shaman) newFireNovaSpell() {
+	var actionId core.ActionID
+
+	actionID.SpellID = 61657
+
+	fireNovaGlyphCDReduction := core.TernaryInt32(shaman.HasMajorGlyph(proto.ShamanMajorGlyph_GlyphOfFireNova), 3, 0)
+	impFireNovaCDReduction := 2 * shaman.Talents.ImprovedFireNova
+	fireNovaCooldown := time.Second * (10 - fireNovaGlyphCDReduction - impFireNovaCDReduction)
+
+	shaman.FireNova = shaman.RegisterSpell(core.SpellConfig{
+		ActionId:    actionID,
+		SpellSchool: core.SpellSchoolFire,
+
+		ResourceType: stats.Mana,
+		BaseCost:     0.22 * shaman.BaseMana,
+
+		Cast: core.CastConfig{
+			DefaultCast: core.Cast{
+				Cost: manaCost,
+				GCD:  core.GCDDefault,
+			},
+			CD: core.Cooldown{
+				Timer:    shaman.NewTimer(),
+				Duration: fireNovaCooldown,
+			},
+		},
+
+		ApplyEffects: core.ApplyEffectFuncAOEDamage(shaman.Env, core.SpellEffect{
+			ProcMask:             core.ProcMaskSpellDamage,
+			BonusSpellHitRating:  float64(shaman.Talents.ElementalPrecision) * 2 * core.SpellHitRatingPerHitChance,
+			BonusSpellCritRating: float64(1.0) * 2 * core.CritRatingPerCritChance,
```

Comment:
I suspect flametongue imbue is handled implisitly?

Reply:
that BonusSpellCritRating line would just give an inherent 2% crit to fire nova, nothing to do with flametongue. Elemental will always have flametongue active but enhance wont and there is no check for the glyph there either. I was trying to handle the crit from the glyph directly in flametongue itself but im not sure it was working, the spellpower was working though

Reply:
@Wumblesnarf This line doesn't handle it?

https://github.com/wowsims/wotlk/blob/master/sim/shaman/weapon_imbues.go#L160

Reply:
It wasnt showing up on the stats view of the actual sim, not sure if its functioning or not. can double check right now, actually

Reply:
it works, and is showing up, i just forgot to have the glyph on. oops.